### PR TITLE
fix(temporal): migrate marker Nexus ids

### PR DESCRIPTION
**Pull request by gregkonush**
## Summary
- migrate legacy marker-backed headerless Nexus operation IDs to the authoritative scheduled event ID
- keep related Nexus cancellation intents aligned with the migrated ID
- add deterministic marker replay regression coverage

## Validation
- `bun test packages/temporal-bun-sdk/tests/workflow/replay.test.ts packages/temporal-bun-sdk/tests/workflow/primitives.test.ts` (28 passed)
- direct TypeScript compilation for `packages/temporal-bun-sdk`
- oxlint on changed files
- `git diff --check origin/main...HEAD`

Follow-up to Codex finding on #12412, discussion `3576877341`.

### CODE REVIEW

**Comment by gregkonush** [comment]
:shipit:

File: packages/temporal-bun-sdk/src/workflow/replay.ts [hunk 1]
```diff
@@ -992,6 +995,81 @@ const commandKindsMatch = (
...
+
+    migratedByScheduledEventId.set(scheduledEventId, operationId)
+    migratedByOperationId.set(entry.intent.operationId, operationId)
+    changed = true
+    return {
+      ...entry,
+      intent: {
+        ...entry.intent,
+        operationId,
+      },
```
[the remark this answers is not in the document]
Comment (gregkonush) [reply packages/temporal-bun-sdk/src/workflow/replay.ts]: Fixed in 63beba1248. Marker migration now applies only to legacy sequence-derived IDs with no recorded Nexus header; caller-supplied operation IDs and their linked cancellation intents remain unchanged. Added a marker-backed explicit operation ID regression; 29 focused tests, TypeScript, oxlint, oxfmt, diff checks, and repository hooks pass.

File: packages/temporal-bun-sdk/src/workflow/replay.ts [hunk 2]
```diff
@@ -992,6 +995,86 @@ const commandKindsMatch = (
...
+  })
+
+  if (migratedByScheduledEventId.size > 0) {
+    commandHistory = commandHistory.map((entry) => {
+      if (entry.intent.kind !== 'request-cancel-nexus-operation') {
+        return entry
+      }
+      const operationId =
+        (entry.intent.scheduledEventId ? migratedByScheduledEventId.get(entry.intent.scheduledEventId) : undefined) ??
+        migratedByOperationId.get(entry.intent.operationId)
```
[the remark this answers is not in the document]
Comment (gregkonush) [reply packages/temporal-bun-sdk/src/workflow/replay.ts]: Fixed in 5105bd0580. Marker migration now preserves cancellation operationId whenever scheduledEventId is already recorded, because the emitted Temporal cancel command is keyed by scheduledEventId. Determinism comparison canonicalizes Nexus cancels by scheduledEventId, so legacy generated IDs and explicit caller IDs both replay correctly. Added regressions for preserved caller IDs and event-scoped equality; 30 focused tests, TypeScript, oxlint, oxfmt, diff checks, and repository hooks pass.

File: packages/temporal-bun-sdk/src/workflow/replay.ts [hunk 3]
```diff
@@ -992,6 +995,82 @@ const commandKindsMatch = (
...
+      scheduledEvent.attributes.value.nexusHeader?.[NEXUS_OPERATION_ID_HEADER]
+    ) {
+      return entry
+    }
+
+    const scheduledEventId = normalizeEventId(scheduledEvent.eventId)
+    const legacySequenceOperationId = `nexus-${entry.intent.sequence}`
+    if (
+      !scheduledEventId ||
+      entry.intent.nexusHeader !== undefined ||
```
[the remark this answers is not in the document]
Comment (gregkonush) [reply packages/temporal-bun-sdk/src/workflow/replay.ts]: Fixed in d47de19e57. Empty nexusHeader maps are now treated as headerless for legacy sequence-derived IDs, while explicit caller IDs remain preserved by the operationId predicate. The migration regression now serializes nexusHeader: {}; 30 focused tests, TypeScript, lint, format, diff checks, and repository hooks pass.

File: packages/temporal-bun-sdk/src/workflow/replay.ts [hunk 4]
```diff
@@ -992,6 +995,82 @@ const commandKindsMatch = (
...
+    ) {
+      return entry
+    }
+
+    const scheduledEventId = normalizeEventId(scheduledEvent.eventId)
+    const legacySequenceOperationId = `nexus-${entry.intent.sequence}`
+    if (
+      !scheduledEventId ||
+      entry.intent.nexusHeader?.[NEXUS_OPERATION_ID_HEADER] !== undefined ||
+      entry.intent.operationId !== legacySequenceOperationId
```
[the remark this answers is not in the document]
Comment (gregkonush) [reply packages/temporal-bun-sdk/src/workflow/replay.ts]: Fixed in 8783003b42. The compatibility path no longer rewrites ambiguous marker IDs. Marker identity is preserved, and worker history extraction maps each headerless scheduled event back to the persisted marker operationId before resolving completions or cancellations. Matching is bounded by the determinism marker event and honors sticky schedule maps, so explicit and legacy nexus-0 IDs both replay deterministically. Added direct identity-map and ambiguous-ID regressions; 32 focused tests and TypeScript pass on current main.

File: packages/temporal-bun-sdk/src/workflow/determinism.ts [hunk 5]
```diff
@@ -371,10 +371,22 @@ const DEFAULT_WORKFLOW_EXECUTION_TIMEOUT_MS = 0
 const DEFAULT_WORKFLOW_ID_REUSE_POLICY = WorkflowIdReusePolicy.ALLOW_DUPLICATE
 const DEFAULT_PARENT_CLOSE_POLICY = ParentClosePolicy.TERMINATE
 
+const normalizeRequestCancelNexusOperationIntent = (intent: WorkflowCommandIntent): WorkflowCommandIntent => {
+  if (intent.kind !== 'request-cancel-nexus-operation' || intent.scheduledEventId === undefined) {
+    return intent
+  }
+  return {
+    ...intent,
+    operationId: `nexus-${intent.scheduledEventId}`,
```
[the remark this answers is not in the document]
Comment (gregkonush) [reply packages/temporal-bun-sdk/src/workflow/determinism.ts:381]: Fixed in 172231efb1. Nexus cancel comparison now canonicalizes both operationId and command id whenever scheduledEventId is present, so replay-reconstructed cancel-nexus-operation-N intents match workflow-emitted request-cancel-nexus-operation-N intents. The regression uses both differing ids and operation ids; 32 focused tests and TypeScript pass after rebasing onto current main.

**Comment by gregkonush** [comment]
@codex review

File: packages/temporal-bun-sdk/src/worker/runtime.ts [hunk 6]
```diff
@@ -1433,8 +1433,13 @@ export class WorkerRuntime {
     try {
       const { results: activityResults, scheduledEventIds: activityScheduleEventIds } =
         await this.#extractActivityResolutions(historyEvents)
-      const { results: nexusResults, scheduledEventIds: nexusScheduleEventIds } =
-        await this.#extractNexusResolutions(historyEvents)
+      const { results: nexusResults, scheduledEventIds: nexusScheduleEventIds } = await this.#extractNexusResolutions(
+        historyEvents,
+        {
+          markerState: historyReplay?.hasDeterminismMarker ? historyReplay.determinismState : undefined,
+          knownScheduleEventIds: stickyEntry?.nexusScheduleEventIds,
```
[the remark this answers is not in the document]
Comment (gregkonush) [reply packages/temporal-bun-sdk/src/worker/runtime.ts]: Fixed in `affcb019d`. Nexus history identity resolution now gives persisted marker/event-derived operation IDs precedence over sticky aliases, so a rejected or drifted sticky cache cannot relabel a headerless completion. Added a regression with a stale sticky alias and a conflicting marker identity; 32 focused replay/primitives tests pass, Oxlint reports zero errors, Oxfmt and both diff checks pass.

[the remark this answers is not in the document]
Comment (gregkonush) [reply packages/temporal-bun-sdk/src/worker/runtime.ts]: Fixed in `e75bb4452`. The worker now accepts cached Nexus schedule aliases only when the sticky entry passed the current history-baseline match; drifted or invalid entries contribute no aliases to extraction or the rebuilt cache. Added a focused selector regression covering accepted and rejected caches; 37 sticky/replay/primitives tests pass, Oxlint has zero errors, Oxfmt and diff checks pass.

[the remark this answers is not in the document]
Comment (gregkonush) [reply packages/temporal-bun-sdk/src/worker/runtime.ts]: Fixed on current rebased head b0942710ab65c725d51bc13a27abc8ea1982e8f2. Sticky Nexus schedule aliases are now accepted only when the sticky cache entry matches the current history baseline; drifted entries pass no aliases into history resolution or the rebuilt cache. Focused validation: 37 Temporal tests passed across sticky-cache, replay, and primitives; TypeScript build passed via the pinned compiler; Oxfmt and `git diff --check origin/main...HEAD` passed.

[the remark this answers is not in the document]
Comment (gregkonush) [reply packages/temporal-bun-sdk/src/worker/runtime.ts]: Fixed on current rebased head b0942710ab65c725d51bc13a27abc8ea1982e8f2. Sticky Nexus schedule aliases are now accepted only when the sticky cache entry matches the current history baseline; drifted entries pass no aliases into history resolution or the rebuilt cache. Focused validation: 37 Temporal tests passed across sticky-cache, replay, and primitives; TypeScript build passed via the pinned compiler; Oxfmt and `git diff --check origin/main...HEAD` passed.
